### ISSUE TITLE
vmagent gce_sd_configs adds a filter when discovering the zone

### ISSUE THREAD

**Issue by mirsafari** [VictoriaMetrics/VictoriaMetrics#3202]
**Describe the bug**
When using vmagent on GCP with `zone: "*"` to automatically discover zones, the discovery of zones fails because the `filter` parameter is passed to the GCP API. It should only be passed when getting a list of compute engine instances and not zones. This affects SD as zones need to be manually defined in the config. 

**To Reproduce**
Deploy vmagent on GCP and define scrape config as follows
```
  vmagent-additional.yaml: |
    - job_name: gce-node-exporter
      gce_sd_configs: 
      - project: "my-test-project"
        zone: "*"
        filter: "labels.vmagent-scrape: true"
        port: 9100
```
Check logs of `vmagent`:
```
2022-10-06T07:44:02.818Z        error   VictoriaMetrics/lib/promscrape/config.go:952    skipping gce_sd_config targets for job_name 
"gce-node-exporter" because of error: cannot get API config: cannot obtain zones for project "my-test-project": cannot obtain zones: 
unexpected status code for "https://compute.googleapis.com/compute/v1/projects/my-test-project/zones?filter=labels.vmagent-
scrape%3A+true"; got 400; want 200; response body: "{\n  \"error\": {\n    \"code\": 400,\n    \"message\": \"Invalid value for field 'filter': 
'labels.vmagent-scrape: true'. Invalid list filter expression.\",\n    \"errors\": [\n      {\n        \"message\": \"Invalid value for field 'filter': 
'labels.vmagent-scrape: true'. Invalid list filter expression.\",\n        \"domain\": \"global\",\n        \"reason\": \"invalid\"\n      }\n    ]\n  
}\n}\n"
```
Replace `zone` to something like `zone: ["europe-west1-d", "europe-west1-c", "europe-west1-b"]` and everything works.

**Expected behavior**
Filter should be used only when getting list of compute instances, not when fetching zone info from `/zones`
**Logs**
```
2022-10-06T07:44:02.818Z        error   VictoriaMetrics/lib/promscrape/config.go:952    skipping gce_sd_config targets for job_name 
"gce-node-exporter" because of error: cannot get API config: cannot obtain zones for project "my-test-project": cannot obtain zones: 
unexpected status code for "https://compute.googleapis.com/compute/v1/projects/my-test-project/zones?filter=labels.vmagent-
scrape%3A+true"; got 400; want 200; response body: "{\n  \"error\": {\n    \"code\": 400,\n    \"message\": \"Invalid value for field 'filter': 
'labels.vmagent-scrape: true'. Invalid list filter expression.\",\n    \"errors\": [\n      {\n        \"message\": \"Invalid value for field 'filter': 
'labels.vmagent-scrape: true'. Invalid list filter expression.\",\n        \"domain\": \"global\",\n        \"reason\": \"invalid\"\n      }\n    ]\n  
}\n}\n"
```

**Version**
```
v1.79.1
```

**Used command-line flags**
```
- -httpListenAddr=:8429
- -promscrape.config=/etc/vmagent/config_out/vmagent.env.yaml
- -promscrape.maxScrapeSize=104857600
- -remoteWrite.maxDiskUsagePerURL=1073741824
- -remoteWrite.tmpDataPath=/tmp/vmagent-remotewrite-data
- -remoteWrite.url=http://prometheus-test.vmagent-test.svc.cluster.local:9090/api/v1/write
```

**Comment by valyala** [comment]
`vmagent` should properly discover GCE zones when `filter` option is set at [gce_sd_configs](https://docs.victoriametrics.com/sd_configs.html#gce_sd_configs) starting from [v1.84.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing this issue as fixed.

**Comment by valyala** [comment]
FYI, the bug should be fixed in [vmagent v1.79.6 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.79.6).
